### DISCUSSION TITLE
How to change columns order? A to C / B to A ..etc

### DISCUSSION THREAD

**Discussion by SaidMoulla** [PHPOffice/PhpSpreadsheet#2497] [Q&A]
Hello, guys!

1/ Can anyone tell me how to change the order of columns using this package?
2/ Where can I find all functions used with this package (like setCellValue/removeColumnByIndex/getMergeCells...)?
3/ How to save multi sheets in one CSV workbook?
4/ How to get the Workbook name?

for the 1st question need like that in VBA:
`
Sub MoveColiD()
      For i = 1 To NbrSheets
        Sheets(i).Columns("G:G").Cut
        Sheets(i).Columns("A:A").Insert Shift:=xlToLeft
      Next I
End Sub
`
This code will move data from Column G to column A without deleting A data (just like inserting new col before the actual A)

Thanks in advance guys!

**Comment by MarkBaker** [comment]
1) Use `insertNewColumnBefore()`; a combination of `rangeToArray()` and `fromArray()` to copy values from the original column into the new column that you've just created; then `removeColumn()` to delete the original column.

2) In the [API Documentation](https://phpoffice.github.io/PhpSpreadsheet/). There is also a lot of useful documentation in [Read the Docs](https://phpspreadsheet.readthedocs.io/en/latest/)

3) Save each worksheet to an individual file, then concatenate the files together.

4) The only workbook name is the filename that you loaded; although there may be additional information stored in the document properties.

**Reply by SaidMoulla** [reply]
Thanks @MarkBaker 

Can you give an example with `rangeToArray()` and `fromArray()` to move Column H data to B (Old B data will become C..) as a test?
I'm a new PhpSpreadsheet user :)

Your help is appreciated ^-^

**Comment by SaidMoulla** [comment]
Hi !

1/ Solved like that:
`// Change columns order for all data / Start N°inClass
$sheet->insertNewColumnBefore("A", 1);
for( $x = 1; $x <= $CountRows ; $x++ ) {
	$GetData = $sheet->getCellByColumnAndRow(8, $x)->getValue();
	$sheet->getCellByColumnAndRow(1, $x)->setValue($GetData);
}
$sheet->removeColumnByIndex(8, 1);`
and so on...

2/ the official documentation is not useful at this point for beginners ^-^
**### 3/ still need help at this point !!!!**
4/ solved like this:
`$Filename = basename( $_FILES["file"]["name"] , ".xlsx" );`
this if using the HTML input to upload file

Hope I can get some help with the 3rd question!!
Regards!
